zerver/views/registration.py

```diff
@@ -73,6 +80,9 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
     key = request.POST['key']
     confirmation = Confirmation.objects.get(confirmation_key=key)
     prereg_user = confirmation.content_object
+    revoked_value = confirmation_settings.STATUS_REVOKED
+    if prereg_user.status == revoked_value:
```

Comment:
Actually, the only status we want to accept here is `STATUS_ACTIVE`. If not, then we show the error page.

Reply:
Coming here with a `STATUS_INACTIVE` invite should also give the user an error page about bad confirmation link.

Reply:
I was super wrong here, I got confused about the various status values here. Their meaning is:
`0` - "inactive", this is what confirmation links are by default.
`STATUS_ACTIVE=1` - confirmation link was succesfully used to register a user.
`STATUS_REVOKED=2` - revoked, can't be used anymore.

So your original change was correct: `if prereg_user.status == confirmation_settings.STATUS_REVOKED: #throw error`  is what we should be doing.